forge/src/executor/builder.rs

```diff
@@ -38,9 +41,25 @@ impl ExecutorBuilder {
         self
     }
 
+    /// Configure the executor's forking mode
+    #[must_use]
+    pub fn with_fork(mut self, url: &str) -> Self {
+        self.fork_url = Some(url.to_string());
+        self
+    }
+
     /// Builds the executor as configured.
     pub fn build(self) -> Executor<EmptyDB> {
-        Executor::new(EmptyDB(), self.config)
+        let db: Box<dyn DatabaseRef> = if let Some(fork) = self.fork_url {
+            let provider = Provider::try_from(fork).unwrap();
+            // TODO: Add pin block
+            // TOOD: Add reading cache from disk
+            let backend = SharedBackend::new(provider, SharedMemCache::default(), None);
+            Box::new(backend)
+        } else {
+            Box::new(EmptyDB())
+        };
+        Executor::new(db, self.config)
```

Comment:
wow this is considerably simpler than what we had for sputnik

Reply:
Still doesn't build - but almost there!